kimchi/src/circuits/polynomials/keccak/circuitgates.rs

```diff
@@ -0,0 +1,211 @@
+//~ The keccak gadget is comprised of _ circuit gates (KeccakXOR, .., and Zero)
```

Comment:
Since this is a general-purpose XOR gate -- nothing about it seems keccak-specific -- doesn't it make sense to separate it out? I know that e.g. it would be handy to use this for SHA256.

Reply:
nice idea

Reply:
Oh well, indeed it has nothing to do with Keccak at this level. It's simply not just any generic XOR, but only 32-bit XOR. So If you're fine with renaming it like Xor32 or similar, that would be nice as well.